### PR TITLE
Inline the binary protos

### PR DESCRIPTION
**Pull request by jchaffraix-slack**
protoc-gen-hack/plugin.go was generating the gzipped binary ProtoDescriptorProto and returning it to protoc so we could save it to a file. protoc was warning us about having those file's content being invalid UTF-8.

Unfortunately newer protoc binary reject files with such a ayload. We do want to keep the binary FileDescriptorProto logic for servers that can send the binary descriptor to their clients but we can inline it in each generated file.

While we don't need to gzip the content anymore, I kept this logic as we were generating large binary strings in Hacklang.

### CODE REVIEW

File: generated/google/protobuf/any_proto.php [hunk 1]
```diff
@@ -101,6 +101,7 @@ public function Name(): string {
   }
 
   public function FileDescriptorProtoBytes(): string {
-    return (string)\gzuncompress(\file_get_contents(\realpath(\dirname(__FILE__)) . '/any_file_descriptor.pb.bin.gz'));
```
Comment (jchaffraix-slack) [inline generated/google/protobuf/any_proto.php:104]: Anything in the `generated/` directory is autogenerated through:
```
bazel run //:gen
```

The diffs are:
1. All the files ending in `.pb.bin.gz` were removed.
2. We inlined their content in the associated `proto.php`.

File: protoc-gen-hack/plugin.go [hunk 2]
```diff
@@ -232,10 +228,7 @@ func writeBinaryFileDescriptor(fdp *desc.FileDescriptorProto, resp *ppb.CodeGene
 		panic(err)
 	}
 
-	// The protoc compiler will complain that this contains invalid UTF-8 data,
-	// but we will ignore that for now.
```
Comment (jchaffraix-slack) [inline protoc-gen-hack/plugin.go:236]: This is the reason we need this PR: the complain is turning into an error in newer version of the `protoc`.

File: protoc-gen-hack/plugin.go [hunk 3]
```diff
@@ -324,7 +317,16 @@ func writeFileDescriptor(w *writer, fdp *desc.FileDescriptorProto) {
...
 	w.ln()
 	w.p("public function FileDescriptorProtoBytes(): string {")
-	w.p("return (string)\\gzuncompress(\\file_get_contents(\\realpath(\\dirname(__FILE__)) . '/%s'));", filepath.Base(binaryFileDescriptorPath(fdp)))
+	// This logic matches what protoc-gen-go plugin does. See this SO thread for an example:
+	// https://stackoverflow.com/questions/60540511/how-to-display-protoc-gen-go-gzipped-filedescriptorproto-as-plaintext
+	binaryProto := generateBinaryFileDescriptor(fdp)
+	binaryProtoStr := ""
+	for i := 0; i < len(binaryProto); i++ {
+		c := binaryProto[i]
+		binaryProtoStr += fmt.Sprintf("\\x%x", c)
```
Comment (jchaffraix-slack) [inline protoc-gen-hack/plugin.go:326]: `%x` prints a single byte in hexadecimal (perfect for the `\x??` format in Hacklang)

**Review by junoatwork — APPROVED** [review]
Thanks for the inline comments!

**Pull request merged**